src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseBuilder.java

```diff
@@ -27,10 +28,11 @@
 	private Client client;
 	@NonNull
 	private ObjectMapper objectMapper;
+	URI uri = URI.create("http://localhost:8080/");
```

Comment:
> As I see it right, this pull request only works for relative uris on the same host at the moment.

I don't understand this point. What do you mean with "only works for relative URIs on the same host"? I would say : of course, therefore they are relative... :-)

The solution is based on the behavior of `URI#resolve` method.

```
       URI.create("http://foo.bar/api")
                .resolve("/api/doc/")
                .equals(
                        URI.create("http://foo.bar/api/doc/"));

        URI.create("http://foo.bar/api") 
                .resolve("/api2/doc/") 
                .equals( 
                        URI.create("http://foo.bar/api2/doc/"));

        URI.create("http://foo.bar/api") //
                .resolve("http://mercateo.com/foo/api/") //
                .equals(
                        URI.create("http://mercateo.com/foo/api/"));
```